### PR TITLE
[sdbus-cpp] New port

### LINKED ISSUE #19025: [New Port Request] sdbus-c++
Library name:
sdbus-c++

Library description:
(From README.md)
sdbus-c++ is a high-level C++ D-Bus library for Linux designed to provide expressive, easy-to-use API in modern C++. It adds another layer of abstraction on top of sd-bus, a nice, fresh C D-Bus implementation by systemd.

Source repository URL:
https://github.com/Kistler-Group/sdbus-cpp/

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by russelltg**
- [sdbus-cpp] new port
- ./vcpkg x-add-version --all

Fixes #19025

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by russelltg** [comment]
This depends on libsystemd, which is why the CI is failing. Two options here:

* Remove from CI
* Add a libsystemd port

Let me know if there's a preference

**Comment by dg0yt** [comment]
CI is mandatory.
libsystemd is proposed in https://github.com/microsoft/vcpkg/pull/31150.

**Comment by russelltg** [comment]
Ok. Converting to draft status for now then

**Comment by jbbjarnason** [comment]
@russelltg the systemd port has been merged, so you could probably continue. 

Side note, if you are not committed to this library I recommend having a look at https://github.com/openbmc/sdbusplus.

File: ports/sdbus-cpp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "${CURRENT_PACKAGES_DIR}/debug/bin"
+)
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/COPYING" "${SOURCE_PATH}/COPYING-LGPL-Exception")
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+
+if ("tool" IN_LIST FEATURES)
+    vcpkg_copy_tools(TOOL_NAMES sdbus-c++-xml2cpp AUTO_CLEAN)
+endif()
+
```
Comment (Adela0814) [inline ports/sdbus-cpp/portfile.cmake:39]: Please remove this line.

File: ports/sdbus-cpp/usage [hunk 2]
```diff
@@ -0,0 +1,3 @@
+sdbus-cpp provides CMake targets:
+    find_package(sdbus-c++ REQUIRED)
+    target_link_libraries(main PRIVATE SDBusCpp::sdbus-c++)
```
Comment (Adela0814) [inline ports/sdbus-cpp/usage:3]: Please add a new line at the end of the file.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**